### PR TITLE
test: add responsive modes to chromatic

### PR DESCRIPTION
**Pull request by codeincontext**
The initial setup for adding responsiveness to storybook. I'm adding [modes](https://www.chromatic.com/docs/modes/) for mobile and desktop

By default Chromatic uses a 1200px width, and that's considered the `1200px` mode. Our existing snapshots have been using this, so I've added a compat option to continue using those snapshots until we're ready for a big switchover

Next steps after this PR:
- For each story, decide on the right modes
- Minimise snapshot changes by continuing with the `1200px` mode for existing stories. Then migrate those to `desktop` later

## Description

- Add [chromatic modes](https://www.chromatic.com/docs/modes/) for mobile and desktop
- Add types for chromatic config in storybook parameters
- Use export buttons as a minimal example

## Testing

See that there are two new mobile snapshots at https://www.chromatic.com/build?appId=672908473f629875f0be294f&number=394

### CODE REVIEW

File: apps/nextjs/.storybook/chromatic.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import "@storybook/csf";
+
+type ChromaticModes = "legacy" | "mobile" | "desktop" | "desktop-wide";
```
[the remark this answers is not in the document]
Comment (codeincontext) [reply apps/nextjs/.storybook/chromatic.ts:3]: I guess we can add it here, I just haven't had a use case for a wider mobile snapshot yet

The viewports are a storybook thing so you can switch between sizes easily in the UI. We can use those named presets in chromatic, but we also aren't required to if we don't need

File: apps/nextjs/.storybook/preview.tsx [hunk 2]
```diff
@@ -28,6 +28,26 @@ const preview: Preview = {
         date: /Date$/i,
       },
     },
+    viewport: {
+      viewports: {
+        mobile: {
+          name: "Mobile",
+          styles: { width: "375px", height: "800px" },
```
[the remark this answers is not in the document]
Comment (codeincontext) [reply apps/nextjs/.storybook/preview.tsx:35]: Yeah, I wasn't sure whether to include a viewport at that size. Maybe worth looking at posthog to see usage. It's another reason why I didn't want to go too deep on multiple mobile snapshot sizes as I'm not sure what the right ones would be

[the remark this answers is not in the document]
Comment (codeincontext) [reply apps/nextjs/.storybook/preview.tsx:35]: I've added mobile-wide now!

File: apps/nextjs/.storybook/chromatic.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (codeincontext) [reply apps/nextjs/.storybook/chromatic.ts]: I've added mobile-wide now!

**Review by mantagen — APPROVED** [review]

**Pull request merged**